src/app/archive/archive_lib/processor.ml

```diff
@@ -111,6 +111,1078 @@ module Public_key = struct
           public_key
 end
 
+module Snapp_state_data = struct
+  let find (module Conn : CONNECTION) (fp : Pickles.Backend.Tick.Field.t) =
+    Conn.find
+      (Caqti_request.find Caqti_type.string Caqti_type.int
+         "SELECT id FROM snapp_state_data WHERE field = ?")
+      (Pickles.Backend.Tick.Field.to_string fp)
+
+  let find_opt (module Conn : CONNECTION) (fp : Pickles.Backend.Tick.Field.t) =
```

Comment:
Could factor out common code of `find/find_opt`, pass in `Conn.find/find_opt` as needed